[16.0][MIG] cetmix_tower_aws: Migration to 16.0

Aldeigja opened the pull request:
Standard migration

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced AWS integration for Cetmix Tower, enabling the use of AWS SDK (Boto3) in server commands.
  - Added support for securely configuring AWS credentials and region within Cetmix Tower.
  - Provided built-in help and documentation for AWS command usage.
  - Included a demo command to list AWS EC2 instances using Python and Boto3.

- **Documentation**
  - Added comprehensive setup, usage, and context guides, including HTML and Markdown documentation.
  - Included a README and translation template for localization support.

- **Tests**
  - Implemented unit tests to verify Boto3 integration and command context behavior.

- **Chores**
  - Updated dependencies to include Boto3.
  - Added manifest and build configuration files.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Aldeigja committed:
[ADD] cetmix_tower_aws: add module

This module integrates the Amazon Web Services (AWS) SDK (boto3) into the
Cetmix Tower, enabling users to manage a wide range of AWS services.
It extends the cx.tower.command  model to provide access to the boto3
Python library, allowing execution of AWS-related commands directly
from the Cetmix Tower interface.

Key features:
- Safely wraps boto3 library for comprehensive AWS service interaction
- Extends command evaluation context with AWS functionality
- Requires boto3 Python dependency

task: 4214

Aldeigja committed:
[FIX] cetmix_tower_aws: fix enumerated list error

- converted enumerated lists in the USAGE.md file to bulleted lists
  This change should resolve the RST syntax error when the README.rst
   is generated

task: 4672

Commit:
[IMP] cetmix_tower_aws: banner and description add

Add the banner and live test url in manifest

Task:4677

Aldeigja committed:
[IMP] cetmix_tower_aws: pre-commit stuff

Aldeigja committed:
[MIG] cetmix_tower_aws: Migration to 16.0

Aldeigja commented on cetmix_tower_aws/pyproject.toml:
```diff
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["whool"]
+build-backend = "whool.buildapi"
```
@coderabbitai We don't need to modify this file as it is maintained by the ci/cd system

Aldeigja commented on cetmix_tower_aws/readme/DESCRIPTION.md:
```diff
@@ -0,0 +1,2 @@
+This module integrates Boto3, the Amazon Web Services (AWS) Software Development Kit (SDK) for Python, into the Cetmix Tower.
+
```
@coderabbitai all blank lines are removed by pre-commit checks

Aldeigja commented on requirements.txt:
```diff
@@ -1,3 +1,4 @@
 # generated from manifests external_dependencies
+boto3
```
@coderabbitai This is standard migration of the Odoo module we don't need make such changes if it works

Aldeigja commented on cetmix_tower_aws/__manifest__.py:
```diff
@@ -0,0 +1,21 @@
+
+{
+    "name": "Cetmix Tower AWS",
+    "summary": """Cetmix Tower AWS EC2 API integration""",
+    "version": "16.0.1.0.0",
```
@coderabbitai This is a standard requirement for the ‘summary’ line of manifest.py in Odoo

Aldeigja commented on cetmix_tower_aws/i18n/cetmix_tower_aws.pot:
```diff
@@ -0,0 +1,54 @@
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 14.0\n"
+"Report-Msgid-Bugs-To: \n"
+"Last-Translator: \n"
+"Language-Team: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
```
@coderabbitai This will be updated by ci/cd system

Aldeigja commented on cetmix_tower_aws/readme/CONFIGURE.md:
```diff
@@ -0,0 +1,39 @@
+
+**Setting up AWS Access**
```
@coderabbitai This structure meets the OCA documentation standards

Aldeigja commented on cetmix_tower_aws/readme/CONFIGURE.md:
```diff
@@ -0,0 +1,39 @@
+
+   To use the AWS integration with Cetmix Tower, you need to create AWS access keys:
+   
+   - Follow the official AWS documentation https://docs.aws.amazon.com/IAM/latest/UserGuide/security-creds.html for creating IAM access keys
```
This is done because links are not rendered in the Odoo App Store.

ivs-cetmix commented on cetmix_tower_aws/readme/CONFIGURE.md:
```diff
@@ -0,0 +1,39 @@
+
```
Lets add the following instruction:

```md
You should install the Python `boto3` package manually if you don't have automatic package installation configured in your Odoo environment. 
Run `pip install boto3` to install it.
```

Aldeigja replied:
Added, thank you for your input.

ivs-cetmix reviewed (changes requested):
Let's add demo data and the documentation update in a separate commit so we could backport it easily.

Aldeigja committed:
[IMP] cetmix_tower_aws: add  AWS demo data

This commit adds demo data for Cetmix Tower AWS module
following the instructions from CONFIGURE.md and USAGE.md. The demo
data includes:

- Secret keys for AWS access credentials
- AWS region variable with  default value set
- Python command for listing EC2 instances using boto3

Also adds installation note to CONFIGURE.md clarifying boto3
dependency requirements.

Task: 4832

Aldeigja commented on cetmix_tower_aws/demo/demo_data.xml:
```diff
@@ -0,0 +1,74 @@
+    <!-- ===============================================
+    Global Variable values
+    ================================================ -->
+    <record id="global_value_aws_region" model="cx.tower.variable.value">
+        <field name="variable_id" ref="variable_aws_region_name" />
+        <field name="value_char">us-east-1</field>
+    </record>
```
@coderabbitai  This is already well described in the official Cetmix Tower documentation

Aldeigja commented on cetmix_tower_aws/demo/demo_data.xml:
```diff
@@ -0,0 +1,74 @@
+    <!-- ===============================================
+    Commands
+    ================================================ -->
+    <record id="command_list_ec2_instances" model="cx.tower.command">
+        <field name="name">List AWS EC2 Instances</field>
+        <field name="action">python_code</field>
+        <field name="code">
+# List EC2 instances using boto3
+result = {"exit_code": 0, "message": None}
+
+session = boto3.Session(
+    aws_access_key_id=#!cxtower.secret.aws_access_key!#,
+    aws_secret_access_key=#!cxtower.secret.aws_secret_access_key!#,
+    region_name={{ aws_region_name }}
+)
+ec2 = session.client('ec2')
+instances = ec2.describe_instances()
+
+instance_details = []
+for reservation in instances['Reservations']:
+    for instance in reservation['Instances']:
+        instance_detail = "Instance ID: " + instance['InstanceId']
+        instance_detail += ", Type: " + instance.get('InstanceType', 'Unknown')
+        instance_detail += ", State: " + instance.get('State', {}).get('Name', 'Unknown')
+        instance_details.append(instance_detail)
+
+if instance_details:
+    result["message"] = "Found " + str(len(instance_details)) + " EC2 instances:\n" + "\n".join(instance_details)
+else:
+    result["message"] = "No EC2 instances found"
+        </field>
+        <field name="access_level">1</field>
+        <field
+            name="tag_ids"
+            eval="[(6, 0, [ref('cetmix_tower_server.tag_custom')])]"
+        />
+        <field name="note">List EC2 instances using boto3 AWS SDK</field>
+    </record>
```
@coderabbitai  This python command is described correctly according to the [examples](https://github.com/cetmix/cetmix-tower/blob/16.0-t4671-mig-cetmix_tower_aws/cetmix_tower_server/demo/demo_data.xml) in the core module

Aldeigja commented on cetmix_tower_aws/__manifest__.py:
```diff
@@ -0,0 +1,24 @@
+    "website": "https://cetmix.com",
+    "live_test_url": "https://cetmix.com/tower",
+    "images": ["static/description/banner.png"],
+    "installable": True,
+    "external_dependencies": {
+        "python": ["boto3"],
+    },
```
This is a migration PR from version 14.0, so we don't need to make major changes, we just need to make the code compatible with the version of Odoo we are migrating to, so we will consider adding this in future PRs

ivs-cetmix commented on cetmix_tower_aws/readme/CONFIGURE.md:
```diff
@@ -1,3 +1,6 @@
+**Prepare Environment**
```
**Prerequisites**

ivs-cetmix reviewed (approved):
LGTM

ivs-cetmix commented on cetmix_tower_aws/demo/demo_data.xml:
```diff
@@ -0,0 +1,74 @@
+    ================================================ -->
+    <record id="variable_aws_region_name" model="cx.tower.variable">
+        <field name="name">AWS Region Name</field>
+        <field name="reference">aws_region_name</field>
```
I think this should be a variable of type "Options" because the number of regions is fixed.

ivs-cetmix reviewed (changes requested):
How about moving the demo data into the regular data? I think those pre-configured values could be really helpful in production too.

ivs-cetmix reviewed (approved):
LGTM

Bearnard21 reviewed (approved):
LGTM

CetmixGitDrone commented:
This PR has the `approved` label and has been created more than 5 days ago. It should therefore be ready to merge by a maintainer (or a PSC member if the concerned addon has no declared maintainer). 🤖

ivs-cetmix commented:
/ocabot merge nobump

CetmixGitDrone commented:
This PR looks fantastic, let's merge it!
Prepared branch [16.0-ocabot-merge-pr-292-by-ivs-cetmix-bump-nobump](https://github.com/cetmix/cetmix-tower/commits/16.0-ocabot-merge-pr-292-by-ivs-cetmix-bump-nobump), awaiting test results.

CetmixGitDrone commented:
Congratulations, your PR was merged at c303be71f24db0edd7ef49dcf2efaf3e94d923c0. Thanks a lot for contributing to cetmix. ❤️
